### ISSUE TITLE
Adding context for journals

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1419]
During user testing, we clearly saw that if we said 'the journal' people doubted we'd really checked, but if we put the journals name we didn't. As mark has correctly raised before, putting in names is tricky as they can be of unknown length, and most are so long that it'll be a pain given how hard we've worked to keep copy short and sweet. 

That said, the journal name is clearly important. So, what I'd propose is that we take the Crossref journal field and apply abbreviations to it.

I found a list of standard abbreviations here: https://www.issn.org/services/online-services/access-to-the-ltwa/ which could serve as the backbone. Annoyingly, there don't seem to be abbreviations for some pretty common words, like 'science', so we'll add those manually (people don't notice, I tested).

I'm hoping that if we spin up a google sheet with the long form of the word to match, and the abbreviation, a find and replace should be quite do-able. Here's the sheet: https://docs.google.com/spreadsheets/d/1lMaOg1zqpAX0GZpJ_d-U9KoNRiTeTy2tLK5lSfJw7E0/edit#gid=1753325698. The twist is that in the standard file, there are like 60k lines, so I'm a bit worried about speed... but my vlookups are still instant so I'm hoping it won't be too bad. Also, 18k of them are N/A, which we can just remove if needed. 

There are other approaches to this, for example using `short-container-title` from crossref also gives short journal names.

_Originally posted by @JosephMcArthur in https://github.com/OAButton/discussion/issues/1381#issuecomment-578490998_

**Comment by markmacgillivray** [comment]
We cache google sheets inside the system anyway and a 60k lookup will be unnoticeably fast on the server. Continuing with this now.

But just to check, you want it to get the crossref short-container-title first and use that if available, but then if not, look up the journal title on the google sheet.

If we don't find a short version for display, do you want the UI to display the long title or just say "the journal"?

**Comment by JosephMcArthur** [comment]
Hmmm, let's use short-container-title first if available, then fallback to our sheet, then 'the journal'

Joseph McArthur
Assistant Director: Right to Research Coalition ( http://righttoresearch.org )
Director / Founder: Open Access Button ( http://openaccessbutton.org )
Twitter: @Mcarthur_Joe ( https://twitter.com/Mcarthur_Joe )
Skype: joseph_mcarthur

**Comment by markmacgillivray** [comment]
Unfortunately this has been taking a while. It is not obvious how we can write an algorithm to translate journal names to short names when they are not available from crossref. 

The LTWA does give a list of words, but they are are actually partials. There is use of a - to seemingly signify whether they are at the start or the end of another word. But there are also some with dashes where it is not clear if that is their purpose.

Also in testing I initially thought that any word that is in the journal name but not in the list should just be included in full in the short version, but that did not match some that I tested. So then I tried only including words that do have a match, but that does not work either.

In the simplest example, the word "science" does not appear in the LTWA, so if I do not include it then the short journal name would be nothing.

However in other cases if I do include words that are not in the LTWA then we get all the generic short words such as "of", "in" etc, and they do not appear in the official short names of some journals I tested.

I am not sure how important this is to you in terms of time it is taking. Getting the short container title from crossref is easy enough and has already been done in the API code and I can finish altering the UI to use it when available, that would take just a few more minutes to complete. But to continue trying to build an algorithm for shortening them, we will need to spend a few more hours or possibly another day or two, and make some decisions about how we want it to operate that have not yet been considered. I couldn't see an explanation on the ISSN website about what their algorithm for using the LTWA is, so spending more time to search for that is also an option, but could take just as long. But if we are going to use that LTWA data we probably should find out whether there is a defined way to apply it to journal name strings or not, otherwise we may just end up showing names that are incorrect and thus reducing the confidence that this issue is supposed to increase.

@JosephMcArthur let me know if you want me to keep working on this or move on to other things.
